### PR TITLE
Feature/trigger roadbook edition

### PR DESCRIPTION
**Pull request by schuetzcarl**
Done in that PR :
- The edition of a DestinationRecord can be triggered by a swipe right on it, which triggers a Dialog screen for editing
- The order of the list can be edited through a Drag & Drop touch action
- Two future slots added in OptionMenu, some future radio buttons to let the user chose which and when he wants to enable the touch action features above.

The remaining DestinationRecord fields to edit in the Dialog, the remove action in the Dialog, and the Dialog link to the "add" floating button concerns the following issue : #53 and will be implemented there.

The radio button task may be done in a next sprint, however it is an easy task.

### CODE REVIEW

**Comment by schuetzcarl** [comment]
For the coverage of the RoadBookFragment class, it is actually not possible to simulate the drag and drop action on the screen.
Hence, the onMove() method of the ItemTouchHelper Callback (In charge to report drag&drop event) is not covered.

However, I found a solution to triggers the back-end data changes as if it were done by some onMove() calls, and then check it the result on the screen is correctly displayed.

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt [hunk 1]
```diff
@@ -132,4 +148,81 @@ class RoadBookFragmentTest {
...
+                    Press.PINPOINT
+                )
+            )
+        )
+    }
+
+    @Test
+    fun swipeRightTriggersEditScreen() {
+        swipeRightTheRecordAt(4)
+        Thread.sleep(4000)
```
Comment (DanielTavaresA) [inline app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt:179]: Could you explain why the Thread.sleep is needed ?

**Review by DanielTavaresA — CHANGES_REQUESTED** [review]
Overall good code. Enought comments to understand what is going on. Some explanation needed about Thread sleep to understand if a cleaner alternative exists.

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt [hunk 2]
```diff
@@ -132,4 +148,81 @@ class RoadBookFragmentTest {
...
+                recyclerView.adapter?.notifyItemMoved(2,3)
+                curr.getRBViewModelForTest().swapRecords(2,2)
+                recyclerView.adapter?.notifyItemMoved(3,4)
+                curr.getRBViewModelForTest().swapRecords(3,3)
+                curr.getRBViewModelForTest().pushSwapsResult()
+            }
+        }
+
+        onView(withText("X17")).check(matches(isDisplayed()))
+        Thread.sleep(4000) // This one is needed
```
[the remark this answers is not in the document]
Comment (schuetzcarl) [reply app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt:225]: Here, clearly to wait that the front end is updated, in order to check the modification triggered by the back-end state change

**Pull request merged**